### ISSUE TITLE
CI rework

### ISSUE THREAD

**Issue by 9il** [libmir/mir-ion#16]
For each of the following repositories we need to setup GitHub CI for the following compiler versions:

 - dmd 2.098.1
 - ldc 12.0.1
 - dmd latest
 - ldc latest
 - dmd beta (allow failures)
 - ldc beta (allow failures)
 - dmd master (allow failures)
 - ldc master (allow failures)

If compilers fails, please open an issue with corresponding link to the CI job.
 
With matrix on:
 - linux 32 bit
 - linux 64 bit
 - windows 64 bit
 - macos (if they are free, needs to check)
 - linux 64 bit on ARM CPU (is it free, needs to check)

Please remove Windows 32 bit support if any.

Repositories to update:

 - [ ] cblas ([DlangScience repo](https://github.com/DlangScience))
 - [ ] lapack
 - [ ] mir-blas
 - [ ] mir-lapack
 - [ ] lubeck ([kaleidicassiciates repo](https://github.com/kaleidicassociates))
 - [x] mir-core
 - [ ] mir-bloomberg
 - [ ] mir-yaml
 - [ ] mir-stat
 - [ ] mir-random
 - [ ] mir-integral
 - [ ] mir-cpuid
 - [ ] mir
 - [x] mir-optim

**Comment by rv32ima** [comment]
macOS is free for OSS projects, but Linux 64-bit on ARM is not available. We would either have to deploy our own self-hosted runner (see: https://docs.github.com/en/actions/hosting-your-own-runners/about-self-hosted-runners) or use CircleCI for this -- I'm fine with either one.

Will phase out 32-bit Windows testing.

Do we want to continue testing with dmd-master? I dropped those jobs, but I guess we can re-enable them.

**Comment by 9il** [comment]
> macOS is free for OSS projects, but Linux 64-bit on ARM is not available. We would either have to deploy our own self-hosted runner (see: https://docs.github.com/en/actions/hosting-your-own-runners/about-self-hosted-runners) or use CircleCI for this -- I'm fine with either one.

Circle CI is better here.

> Do we want to continue testing with dmd-master?

Yes.

**Comment by rv32ima** [comment]
@9il I have a question about this part:
> If compilers fails, please open an issue with corresponding link to the CI job.

Do you mean that you want to have issues automatically opened upon CI job failures?

**Comment by 9il** [comment]
> Do you mean that you want to have issues automatically opened upon CI job failures?

No, just a common issue manually opened. So we can track what is going on wrong.
